docs/how-tos/hello-world-module.md

```diff
@@ -0,0 +1,508 @@
+
+1. Run the test script in your terminal:
+
+    {{< tabs >}}
+{{% tab name="Python" %}}
+
+It's best practice to use a virtual environment for running Python scripts.
+You'll also need to install the dependency Pillow in the virtual environment before running the test script.
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+source .venv/bin/activate
+pip install Pillow
+python3 test.py
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+    The image you saved should open on your screen, and a random number should print to your terminal.
+
+## Choose an API to implement
+
+Now it's time to decide which Viam [APIs](/appendix/apis/#component-apis) make the most sense for your module.
+You need a way to return an image, and you need a way to return a number.
+
+If you look at the [camera API](/appendix/apis/components/camera/), you can see the `GetImage` method, which returns an image.
+That will work for the image.
+None of the camera API methods return a number though.
+
+Look at the [sensor API](/appendix/apis/components/sensor/), which includes the `GetReadings` method.
+You can return a number with that, but the sensor API can't return an image.
+
+Your module can support multiple modular resources, so let's make two modular resources: a camera to return the image, and a sensor to return a random number.
+
+{{% alert title="Note" color="note" %}}
+
+For a quicker hello world experience, you can skip the sensor and only create a camera modular resource.
+If you prefer the simpler path, skip the sensor sections in the steps below.
+
+{{% /alert %}}
+
+## Generate stub files
+
+The easiest way to generate the necessary files for your module is to use the [Viam CLI](/cli/).
+
+### Generate the camera files
+
+The CLI module generator generates all the necessary files for one modular resource at a time.
+First let's generate the camera component files, and we'll add the sensor code later.
+
+1. Run the `module generate` command in your terminal:
+
+    ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+    viam module generate
+    ```
+
+1. Follow the prompts, selecting the following options:
+
+   - Module name: `hello-world`
+   - Language: Your choice
+   - Visibility: `Private`
+   - Namespace/Organization ID:
+     - In the [Viam app](https://app.viam.com), navigate to your organization settings through the menu in upper right corner of the page.
+       Find the **Public namespace** and copy that string.
+       In the example snippets below, the namespace is `jessamy`.
+   - Resource to add to the module (API): `Camera Component`.
+   We will add the sensor later.
+   - Model name: `hello-camera`
+   - Enable cloud build: `No`
+   - Register module: `No`
+
+1. Hit your Enter key and the generator will generate a folder called <file>hello-world</file> containing stub files for your modular camera component.
+
+### Generate the sensor code
+
+{{< expand "Click if you are also creating a sensor component" >}}
```

Comment:
The copy pasting is a bit awkward and I don't love it--it makes the whole thing twice as long, and makes it look like creating a basic module is much more complex than it actually is. I do actually mean to position it as "you'll create a camera resource and if you want to you can also create a sensor," but I didn't fully commit in the intro since I'd already written it one way and wasn't sure what you'd think of the optionality.

I think I should just change the intro and keep this guide shorter the way it is. Thoughts?

Reply:
*the way that it is in terms of optionality, whether or not the sensor code all gets more simply copied per your other comment.